### PR TITLE
Aastha_Assignment_17_Done_60mins

### PR DESCRIPTION
**Pull request by AasthaJJun20**
->2 classes pushed : 1)CountCharsOfString & FindIndexOfChars
->CountCharsOfString took me hardly 10-15mins but the second requirement was tough :(
->PFB screenshots:

CountCharsOfString-output
FindIndexOfChars-output

### CODE REVIEW

File: src/aasthaJJun20/FindIndexOfChars.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package aasthaJJun20;
+
+public class FindIndexOfChars {
+	void countChars(String input){
+		for(int index = 0; index < input.length(); index++){
+			int count = 0;
+			//char ch = input.charAt(index);
+			if(index == input.indexOf(input.charAt(index))){
+				for(int innerIndex = 0; innerIndex < input.length(); innerIndex++){
```
Comment (KrishnaKTechnocredits) [inline src/aasthaJJun20/FindIndexOfChars.java]: Dear, here two for loop is not required. One is sufficient. Please optimize your code.

File: src/aasthaJJun20/CountCharsOfString.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+			if(Character.isDigit(input.charAt(index)))
+				countDigit++;
+			else if(Character.isUpperCase(input.charAt(index)))
+					countUpperCase++;
+			else if(Character.isLowerCase(input.charAt(index)))
+				countLowerCase++;
+			else
+				countSpecialChar++;
+		}
+		System.out.println("Total digit: "+countDigit+"\nTotal Letters: "+(countUpperCase+countLowerCase)+"\nTotal Uppercase: "+countUpperCase+"\nTotal Lowercase: "+countLowerCase+"\nTotal Special char: "+countSpecialChar);
```
Comment (KrishnaKTechnocredits) [inline src/aasthaJJun20/CountCharsOfString.java:16]: No body able to crack this logic but you did it, Kudos to you !!! Keep it up.

File: src/aasthaJJun20/FindIndexOfChars.java (hunk 1 above)
Comment (AasthaJJun20) [reply src/aasthaJJun20/FindIndexOfChars.java]: Changes done as per you comment. please review. :)

File: src/aasthaJJun20/FindIndexOfChars.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package aasthaJJun20;
+
+public class FindIndexOfChars {
+	void countChars(String input){
+		for(int index = 0; index < input.length(); index++){
+			if(index == input.indexOf(input.charAt(index))){
+				if(input.indexOf(input.charAt(index)) != input.lastIndexOf(input.charAt(index))){
```
Comment (KrishnaKTechnocredits) [inline src/aasthaJJun20/FindIndexOfChars.java:7]: Yes, This is expected, Good job Dear.

**Review by KrishnaKTechnocredits — APPROVED** [review]

**Pull request merged**